internal/pkg/service/stream/storage/statistics/repository/stats_allocation.go

```diff
@@ -27,41 +28,62 @@ func (r *Repository) estimateSliceSizeOnSliceOpen() {
 		// We need to update the value to the Slice entity before saving,
 		// before the callback is completed, because later the entity value is already stored
 		// in the WRITE phase transaction and cannot be modified.
-		if err := r.estimateSliceSize(file, slice).Do(ctx).Err(); err != nil {
+		if err := r.estimateSliceSize(ctx, file, slice); err != nil {
 			// Error is not fatal
 			r.logger.Errorf(ctx, `cannot calculate slice pre-allocated size: %s`, err)
 		}
 		return nil
 	})
 }
 
-func (r *Repository) estimateSliceSize(file model.File, slice *model.Slice) *op.TxnOp[datasize.ByteSize] {
-	return r.
-		maxUsedDiskSizeBySliceIn(slice.SinkKey, recordsForSliceDiskSizeCalc).
-		OnResult(func(r *op.TxnResult[datasize.ByteSize]) {
-			slice.LocalStorage.AllocatedDiskSpace = file.LocalStorage.DiskAllocation.ForNextSlice(r.Result())
-		})
+func (r *Repository) estimateSliceSize(ctx context.Context, file model.File, slice *model.Slice) error {
+	// Get maximum slice size
+	size, err := r.maxUsedDiskSizeBySliceIn(ctx, slice.SinkKey, recordsForSliceDiskSizeCalc)
+	if err != nil {
+		return err
+	}
+
+	// Calculate allocated disk space for the new slice
+	slice.LocalStorage.AllocatedDiskSpace = file.LocalStorage.DiskAllocation.ForNextSlice(size)
+	return nil
 }
 
 // maxUsedDiskSizeBySliceIn scans the statistics in the parentKey, scanned are:
 //   - The last <limit> slices in level.LevelStaging (uploaded slices).
 //   - The last <limit> slices in level.LevelTarget  (imported slices).
-func (r *Repository) maxUsedDiskSizeBySliceIn(parentKey fmt.Stringer, limit int) *op.TxnOp[datasize.ByteSize] {
+func (r *Repository) maxUsedDiskSizeBySliceIn(ctx context.Context, parentKey fmt.Stringer, limit int) (datasize.ByteSize, error) {
+	// Get last <limit> slices from staging and target levels
+	var lastStagingSlices, lastTargetSlices []model.Slice
+	listSlicesOpts := []iterator.Option{iterator.WithSort(etcd.SortDescend), iterator.WithLimit(limit)}
+	err := op.Txn(r.client).
+		Then(r.storage.Slice().ListInLevel(parentKey, model.LevelStaging, listSlicesOpts...).WithAllTo(&lastStagingSlices)).
+		Then(r.storage.Slice().ListInLevel(parentKey, model.LevelTarget, listSlicesOpts...).WithAllTo(&lastTargetSlices)).
+		Do(ctx).
+		Err()
+	if err != nil {
+		return 0, err
+	}
+
+	// Load and process statistics for each slice from the previous step
```

Comment:
The main issue was, wit the `estimateSliceSize/maxUsedDiskSizeBySliceIn`.
- Individual records in the etcd prefix are no more full slice statistics, but only partial statistics from a source node.
- So I cannot get easily the last `10` slice statistics.
- I must load the last `10` slices from the storage repository, and then sum statistics for there slices.